### ISSUE TITLE
Update install-windows-prereqs.ps1 to pull Intel Windows SGX packages from cache

### ISSUE THREAD

**Issue by CodeMonkeyLeet** [openenclave/openenclave#3015]
One of the issues causing instability in the contributor/user setup instructions on a Windows device is the fact that the install-windows-prereqs.ps1 helper script targets the Intel Developer Zone download links to acquire the SGX dependencies on Windows. This has the following issues:

* Intel Developer Zone does not provide the ability to target a specific instance of the the target package by instance, so updating of a package breaks links to old instances.
* These links are usually CDN links and are brittle, breaking when Intel pushes updates to the file without re-versioning it (for example, to fix errors in the published package).

The mitigate these issues, we should host the specific version dependencies of the packages from a stable endpoint, which will also allow us to maintain multiple versions of the packages (e.g. one for latest v0.x.x release branch, and another for master branch).

* Create a public storage container in a cloud provider and cache the required versions of the Intel Windows SGX packages there.
* Per legal review, we will also need to update docs around use of the install-windows-prereqs.ps1 script and its help text to indicate that its use implies consent to the user agreement to the Intel EULA for those packages.

**Comment by radhikaj** [comment]
Francisco will add links to WU Intel packages for WS2016 and WS2019/Windows 10.

**Comment by fchinchilla** [comment]
# Server 2016
* Base INF
  * Generic Search https://www.catalog.update.microsoft.com/Search.aspx?q=SgxLCDevice . If you click on any of the downloads there is a link to download a specific file in the format GUID_SHA1HASH.cab
  * 2.7.100 http://download.windowsupdate.com/d/msdownload/update/driver/drvs/2020/04/8e78fd6b-efeb-4952-ab0d-945e61c164ba_f46d7480dc60858df4291eb20cc088183e6d09c7.cab
* PSW
  * Not currently available on WU
* DCAP
  * Generic Search https://www.catalog.update.microsoft.com/Search.aspx?q=SgxLCDevice_DCAP
  * 1.6 http://download.windowsupdate.com/d/msdownload/update/driver/drvs/2020/05/08cc8440-9f38-4635-9685-cdbf476666fa_92188002588d6b5fa58814bc92b78ad33ddf11af.cab 

# Server 2019
* Base INF
  * Generic Search https://www.catalog.update.microsoft.com/Search.aspx?q=ACPI\INT0E0C
  * http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2020/04/da362676-240a-4ec6-98cf-4cc4430c84be_cdf84e14313b8d9aa09a624a03ad6c6c55983021.cab 

* PSW
  * https://www.catalog.update.microsoft.com/Search.aspx?q=SWC%5cVEN_INT%26DEV_0E0C
  * 2.7.101 http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2020/05/48e7c1e9-6de8-46ee-8ff9-46aa7b7ee5b9_ab5277f79b04b450e6e7c3a624d99c1b1722e6f2.cab
* DCAP
  * https://www.catalog.update.microsoft.com/Search.aspx?q=SWC%5cVEN_INT%26DEV_0E0C_DCAP
  * 1.6.100 http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2020/05/4f1f1691-c4b8-422c-9ca9-d22ebee726cc_a95a9c0e06caed691d2cbecae41bb0180b5a5fb5.cab

**Comment by BRMcLaren** [comment]
Done task 1 as part of: https://github.com/openenclave/openenclave/pull/3498

**Comment by rs--** [comment]
@radhikaj given that the technical work is already done, is there anything left for @BRMcLaren or myself to do for the ticket?

**Comment by BRMcLaren** [comment]
@rs-- Yeah there is, as the install scripts are still pulling from the IDZ.

https://github.com/openenclave/openenclave/blob/master/scripts/install-windows-prereqs.ps1#L17

This needs to be changed so it pulls from a persistent storage location (or WU once the PSW/DCAP merge is complete). Then the scripts need to be modified to include the legal components outlined here

> Per legal review, we will also need to update docs around use of the install-windows-prereqs.ps1 script and its help text to indicate that its use implies consent to the user agreement to the Intel EULA for those packages.
Specifically, this would be Exhibit A in the PSW_INF_RS3_and_above\component\Win Server 2019\926b5eea-853d-4ec9-bb71-b8a6bce56f3a\INTEL Software License Agreement 11.2.17.pdf contained in the Intel PSW for Windows package.
